Makefile

```diff
@@ -4,27 +4,22 @@ exe = github.com/aelsabbahy/goss/cmd/goss
 pkgs = $(shell ./novendor.sh)
 cmd = goss
 TRAVIS_TAG ?= "0.0.0"
-GO_FILES = $(shell find . \( -path ./vendor -o -name '_test.go' \) -prune -o -name '*.go' -print)
+GO_FILES = $(shell git ls-files -- '*.go' ':!:*vendor*_test.go')
 GO111MODULE=on
 
 .PHONY: all build install test release bench fmt lint vet test-int-all gen centos7 wheezy precise alpine3 arch test-int32 centos7-32 wheezy-32 precise-32 alpine3-32 arch-32
 
-all: test-all dgoss-sha256
+all: test-short-all test-int-all dgoss-sha256
 
-test-all: fmt lint vet test test-int-all
```

Comment:
I introduced `test-short-all` to represent all the short tests like lints and unit-tests, which can run on each platform. 

The integration tests can't run on non-linux unless we come up with a different approach, but I definitely didn't want to include that scope here.

Reply:
Honestly, I give up on trying to have a proper integration test for Windows/OSX

Solid unit tests + a small smoke test might be sufficient.

Reply:
Yeah, 👍. If we think it becomes worthwhile to do more than unit-test, I reckon it'll be reasonably straightforward to achieve via `TestMain` and blackbox'ing against some golden files.

Reply:
Curious on your Test mail suggestion. Whenever you have a chance (no rush). Can you link me a psudocode gist example of what you mean.

On a possibility related note. There's a small part of me that's wondering if it would be crazy to try to test goss with goss. :mindblown: some changes would need to happen to goss first to make it possible.